ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,6 +6,141 @@ set -o pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
+trap 'FRC=$?; createHeterogeneousJunit; debug' EXIT TERM
+
+# Print failed node, co information for debug purpose
+function debug() {
+    if (( FRC != 0 )); then
+        echo -e "Describing abnormal nodes...\n"
+        oc get node --no-headers | awk '$2 != "Ready" {print $1}' | while read node; do echo -e "\n#####oc describe node ${node}#####\n$(oc describe node ${node})"; done
+        echo -e "Describing abnormal operators...\n"
+        oc get co --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read co; do echo -e "\n#####oc describe co ${co}#####\n$(oc describe co ${co})"; done
+    fi
+}
+
+# Generate the Junit for migration
+function createHeterogeneousJunit() {
+    echo "Generating the Junit for heterogeneous"
+    if (( FRC == 0 )); then
+      cat >"${ARTIFACT_DIR}/junit_heterogeneous.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="heterogeneous" tests="1" failures="0">
+  <testcase classname="heterogeneous" name="heterogeneous test should succeed"/>
+</testsuite>
+EOF
+    else
+      cat >"${ARTIFACT_DIR}/junit_heterogeneous.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="heterogeneous" tests="1" failures="1">
```

Comment:
@aleskandro the step `ipi-install-heterogeneous` is not only for arch migration, and we already had a class for `cluster arch migration` [here](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/arch-migration/cucushift-upgrade-arch-migration-commands.sh#L28-L29), so if I checked to the below ok? or do you have better a suggestion?
```
name="cluster_multi_arch_nodes"
classname="cluster multi arch nodes"
name="Adding secondary arch nodes to multi-arch cluster should succeed"
```

Reply:
I would like to have a junit in any case, and yes, to avoid the two signals overwrite each other is a good point... let's discuss the other things in the other thread below.